### PR TITLE
Use buildkite to drive slurm cpu/gpu tests

### PR DESCRIPTION
**Pull request by jakebolewski**
# Description

Switch cluster tests to use Buildkite https://buildkite.com/clima for dynamic CI tests that need to run on CPU / GPU cluster resources.

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by simonbyrne** [comment]
I triggered a job here: https://buildkite.com/clima/climatemachine-ci/builds/44

**Comment by simonbyrne** [comment]
if that works, i think we just need to change the `bors.toml` to use `buildkite/climatemachine-ci` instead of slurmci

**Comment by simonbyrne** [comment]
This is looking more promising:
https://buildkite.com/clima/climatemachine-ci/builds/50

Can you also update the pipeline.yml file to add a `key` to each step (which can match the `label` for now)? We can then use that for our output directory (rather than the slurm id)

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,847 @@
+env:
+  OPENBLAS_NUM_THREADS: 1
+  CLIMATEMACHINE_SETTINGS_FIX_RNG_SEED: "true"
+
+steps:
+  - name: "cpu-advection_diffusion_model_1dimex_bgmres"
```
Comment (simonbyrne) [inline .buildkite/pipeline.yml:6]: and lets use either dashes or underscores (I don't really care which).

**Comment by jakebolewski** [comment]
bors try

**Comment by jakebolewski** [comment]
bors try-

**Comment by jakebolewski** [comment]
bors try

**Comment by jakebolewski** [comment]
bors try

**Comment by jakebolewski** [comment]
bors try-

**Comment by jakebolewski** [comment]
bors try

**Comment by jakebolewski** [comment]
bors try-

**Comment by jakebolewski** [comment]
bors try

**Comment by jakebolewski** [comment]
bors try-

**Comment by jakebolewski** [comment]
bors try

**Comment by simonbyrne** [comment]
any idea what's wrong with the windows CI?

**Comment by jakebolewski** [comment]
unfortunately no, my branch was a bit behind master so I've updated it just to check if that addresses the issue.

it seems like maybe an upstream problem in one of the packages (Polynomials?).  Linux worked fine though

**Comment by jakebolewski** [comment]
@simonbyrne  it's an upstream issue:
https://github.com/JuliaMath/Polynomials.jl/issues/248#issue-673029252

**Comment by simonbyrne** [comment]
It's not clear. Could also be this: https://github.com/JuliaLang/Pkg.jl/issues/1942

**Comment by jakebolewski** [comment]
I think since the redirect is to master and master was the issue on `Polynomials` (not the release candidate) that seems to be it :grimacing:

**Comment by simonbyrne** [comment]
Farewell @climabot!

**Pull request merged**